include/zephyr/kernel.h

```diff
@@ -4002,6 +4002,11 @@ struct k_work_queue_config {
 	 * control.
 	 */
 	bool no_yield;
+
+	/** Control whether the work queue thread should be marked as
+	 * essential thread.
+	 */
+	bool essential;
```

Comment:
instead of making `essential` a global config for all work queues, it makes more sense to have a 'system' config, indicating this is a system work queue, then based on this, you can set the essential flag on system work queues.

Reply:
> instead of making essential a global config for all work queues, it makes more sense to have a 'system' config, indicating this is a system work queue, then based on this, you can set the essential flag on system work queues.

you mean to have something like:
```c
/* new type in addition to `k_work_queue_config` */
struct k_sys_work_queue_config {
	const char *name;
	bool no_yield;
	bool essential;
};

void k_sys_work_queue_start(struct k_work_q *sys_queue,
			k_thread_stack_t *stack,
			size_t stack_size,
			int prio,
			const struct k_sys_work_queue_config *cfg)
{
	struct k_work_queue_config cfg = {
		.name = cfg->name,
		.no_yield = cfg->no_yield
	};
	k_work_queue_start(sys_queue, stack, stack_size, prio)
	sys_queue.thread.base.user_options |= K_ESSENTIAL;
}
```



But I would like to say that the idea behind including `essential` in the queue config is to allow the user to mark their own queue also as essential not only system work queue. for example we have a system where it uses a custom Bluetooth transport which has its own workq and if this workq thread crashed it doesn't make sense for our system to continue to work. WDYT ?

Reply:
no, I mean

```
/* new type in addition to `k_work_queue_config` */
struct k_sys_work_queue_config {
	const char *name;
	bool no_yield;
	bool system_wq;
};
```

and later on: 

```
 k_thread_create(&queue->thread, ...., prio, cfg->system_wq ? K_ESSENTIAL : 0, K_FOREVER);
```

> But I would like to say that the idea behind including `essential` in the queue config is to allow the user to mark their own queue also as essential not only system work queue.

that is not what the title/description of the PR says...

Reply:
Isn't there only 1 system work queue? Essential might be the better term 🤷🏼‍♂️

Reply:
> and later on:
 `k_thread_create(&queue->thread, ...., prio, cfg->system_wq ? K_ESSENTIAL : 0, K_FOREVER);`

That means `bool system_wq` will be added to `k_work_queue_config` type without creating the new type `k_**sys**_work_queue_config`, otherwise we will need to have 2 implementations of `k_work_queue_start()`

Reply:
> Isn't there only 1 system work queue? Essential might be the better term 🤷🏼‍♂️

there is only one, and hence the suggestion by only reading the PR description and the related issue, however, there is another commit in this PR that is not mentioned in the PR description which allows essential to be set on any work queue. I was commenting based on the PR description.